components/script/dom/websocket.rs

```diff
@@ -353,9 +378,10 @@ impl WebSocketMethods for WebSocket {
         let mut other_sender = self.sender.borrow_mut();
         let my_sender = other_sender.as_mut().unwrap();
 
-        self.buffered_amount.set(self.buffered_amount.get() + (data.0.as_bytes().len() as u32));
+        self.buffered_amount.set(bufferedAmount);
```

Comment:
I would suggest leaving the self.buffered_amount.get() in this calculation, and making callers pass in only the new amount to increase it by.